### PR TITLE
Provide module shards Config to Controller instead of tmp file

### PR DESCRIPTION
**Pull request by PeterSuna**
Provide custom moduleShardsConfig in controller instead of storing
moduleShardsConfig in tmp file: "/tmp/module-shards.conf".
This file was later read in initializing class ConfigurationImpl.

### CODE REVIEW

File: lighty-core/lighty-clustering/pom.xml [hunk 1]
```diff
@@ -36,6 +36,10 @@
             <groupId>org.opendaylight.controller</groupId>
             <artifactId>sal-cluster-admin-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.opendaylight.controller</groupId>
+            <artifactId>sal-distributed-datastore</artifactId>
+        </dependency>
         <dependency>
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-core/lighty-clustering/pom.xml:43]: ignore

File: lighty-core/lighty-clustering/pom.xml [hunk 2]
```diff
@@ -36,6 +36,10 @@
             <groupId>org.opendaylight.controller</groupId>
             <artifactId>sal-cluster-admin-api</artifactId>
         </dependency>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-core/lighty-clustering/pom.xml:39]: ignore

File: lighty-core/lighty-controller/pom.xml [hunk 3]
```diff
@@ -49,19 +49,6 @@
...
-        <dependency>
-            <groupId>org.opendaylight.infrautils</groupId>
-            <artifactId>metrics-impl</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>org.opendaylight.infrautils</groupId>
-            <artifactId>metrics-api</artifactId>
-        </dependency>
-        <!--odl-yangtools-common-->
         <dependency>
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-core/lighty-controller/pom.xml:52]: ignore

File: lighty-models/openconfig-models/pom.xml [hunk 4]
```diff
@@ -31,7 +31,7 @@
         </dependency>
         <dependency>
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-models/openconfig-models/pom.xml:32]: ignore

File: lighty-core/lighty-clustering/src/main/java/io/lighty/core/cluster/kubernetes/KubernetesClusteringHandlerImpl.java [hunk 5]
```diff
@@ -79,16 +74,10 @@ public void initClustering() {
...
-                String data = ClusteringConfigUtils.generateModuleShardsForMembers(memberRoles);
-                Files.write(Paths.get(ClusteringConfigUtils.MODULE_SHARDS_TMP_PATH),
-                        data.getBytes(StandardCharsets.UTF_8));
-                this.moduleShardsConfig = Optional.of(ClusteringConfigUtils.MODULE_SHARDS_TMP_PATH);
-                return;
-            } catch (IOException e) {
-                LOG.info("Tmp module-shards.conf file was not created - error received {}", e.getMessage());
-            }
+            List<String> memberRoles = akkaDeploymentConfig.getStringList("akka.cluster.roles");
+            String data = ClusteringConfigUtils.generateModuleShardsForMembers(memberRoles);
```
Comment (ihrasko) [inline lighty-core/lighty-clustering/src/main/java/io/lighty/core/cluster/kubernetes/KubernetesClusteringHandlerImpl.java]: final

File: lighty-core/lighty-clustering/src/main/java/io/lighty/core/cluster/kubernetes/KubernetesClusteringHandlerImpl.java [hunk 6]
```diff
@@ -79,16 +74,10 @@ public void initClustering() {
...
-                List<String> memberRoles = akkaDeploymentConfig.getStringList("akka.cluster.roles");
-                String data = ClusteringConfigUtils.generateModuleShardsForMembers(memberRoles);
-                Files.write(Paths.get(ClusteringConfigUtils.MODULE_SHARDS_TMP_PATH),
-                        data.getBytes(StandardCharsets.UTF_8));
-                this.moduleShardsConfig = Optional.of(ClusteringConfigUtils.MODULE_SHARDS_TMP_PATH);
-                return;
-            } catch (IOException e) {
-                LOG.info("Tmp module-shards.conf file was not created - error received {}", e.getMessage());
-            }
+            List<String> memberRoles = akkaDeploymentConfig.getStringList("akka.cluster.roles");
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-core/lighty-clustering/src/main/java/io/lighty/core/cluster/kubernetes/KubernetesClusteringHandlerImpl.java]: done

**Review by ihrasko — APPROVED** [review]

File: lighty-core/lighty-clustering/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-core/lighty-clustering/pom.xml:39]: ignore

**Review by ihrasko — APPROVED** [review]

**Pull request merged**